### PR TITLE
Allow plotting service to go find files

### PR DESCRIPTION
**Pull request by Pasarus**
Closes None, fixes an issue in staging and prod

## Description
Better handles the finding file race condition on staging, the previous PR had an issue where it relied on a hidden race condition that was exposed by some seemingly basic refactoring in the code to fix the e2e tests. This race condition is now handled much better.

### CODE REVIEW

**Review by Dagonite — APPROVED** [review]
LGTM code-wise.

**Comment by Pasarus** [comment]
Issue made to reimplement the commented out workflow:
[#217](https://github.com/fiaisis/plotting-service/issues/217)

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -36,28 +36,28 @@ jobs:
...
+#          working-directory: data-viewer
+#          headed: true
+#          browser: firefox
+#
+#      - name: Upload screenshots
+#        uses: actions/upload-artifact@v4
+#        if: failure()
+#        with:
+#          name: cypress-snapshots
+#          path: data-viewer/cypress/screenshots
```
Comment (Dagonite) [inline .github/workflows/test.yml:63]: Commented out YML code instead of deleted. This right?

File: data-viewer/src/components/NexusViewer.tsx [hunk 2]
```diff
@@ -34,56 +36,59 @@ export default function NexusViewer(props: {
...
+    const loadedToken = localStorage.getItem("scigateway:token") ?? ""
+    setToken(loadedToken);
+    setApiUrl(props.apiUrl.includes("localhost") ? props.apiUrl : `${window.location.protocol}//${window.location.hostname}/plottingapi`)
+
+    const fileQueryUrl = `${props.apiUrl}/find_file/instrument/${props.instrument}/experiment_number/${props.experimentNumber}`
+    const fileQueryParams = `filename=${props.filename}`;
     const headers: { [key: string]: string } = {'Content-Type': 'application/json'};
-    if (token != "") {
-      headers['Authorization'] = `Bearer ${token}`;
+    console.log("Token set as: " + loadedToken)
```
Comment (Dagonite) [inline data-viewer/src/components/NexusViewer.tsx:53]: Should this log be here?

**Review by Dagonite — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (Dagonite) [reply .github/workflows/test.yml:63]: Oh, just seen above.

File: data-viewer/src/components/NexusViewer.tsx (hunk 2 above)
Comment (Pasarus) [reply data-viewer/src/components/NexusViewer.tsx]: Good catch here.

**Review by Dagonite — APPROVED** [review]
LGTM

**Pull request merged**